pkgs/development/python-modules/craft-application_1/default.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, nix-update-script
+, python3
+}:
+
+buildPythonPackage rec {
+  pname = "craft-application_1";
+  version = "1.2.1";
+
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "canonical";
+    repo = "craft-application";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-CXZEWVoE66dlQJp4G8tinufjyaDJaH1Muxz/qd/81oA=";
+  };
+
+  postPatch = ''
+    substituteInPlace craft_application/__init__.py \
+      --replace-fail "dev" "${version}"
+
+    substituteInPlace pyproject.toml \
+      --replace-fail "setuptools==67.7.2" "setuptools"
+  '';
+
+  propagatedBuildInputs = with python3.pkgs; [
+    craft-cli
+    craft-parts
+    craft-providers
+    pydantic-yaml_0
+    pyyaml
+    setuptools
+    setuptools-scm
+  ];
+
+  env.SETUPTOOLS_SCM_PRETEND_VERSION = version;
```

Comment:
```suggestion
```
This shouldn't be needed anymore, see https://github.com/NixOS/nixpkgs/commit/8056f9250ce8b7f9bdf0ccdc6c7a6808ae8f1be9